azext_edge/tests/edge/init/int/test_init_int.py

```diff
@@ -137,6 +143,194 @@ def test_init_scenario(
+    assert instance_props["schemaRegistryNamespace"] == schema_registry_namespace
+
+    expected_components = {"adr", "akri", "connectors", "dataflows", "schemaRegistry"}
+    disabled_components = []
+    unexpected_components = []
+    for component, state in instance_props["components"].items():
+        if state["state"].lower() != "enabled":
+            disabled_components.append(component)
+        if component in expected_components:
+            expected_components.remove(component)
+        else:
+            unexpected_components.append(component)
+
+    error_msg = []
+    if disabled_components:
+        error_msg.append(f"The following components are disabled: {disabled_components}.")
+    if unexpected_components:
+        error_msg.append(f"The following components are unexpected: {unexpected_components}.")
+    if expected_components:
+        error_msg.append(f"The following components are missing: {expected_components}.")
+    if error_msg:
+        raise AssertionError("\n".join(error_msg))
+
+    tree = run(f"az iot ops show -n {instance_name} -g {resource_group} --tree")
+    # no resource sync rules if disable rsync rules
+    assert ("adr-sync" not in tree) is disable_rsync_rules
+    assert instance_name in tree
+    assert expected_custom_location in tree
+    assert "azure-iot-operations-platform" in tree
+
+    # list failed to return collection response YAY
+    # instance_rg_list = run(f"az iot ops list -g {resource_group}")
+    # assert instance_name in [inst["name"] for inst in instance_rg_list]
+    # instance_sub_list = run("az iot ops list")
+    # assert instance_name in [inst["name"] for inst in instance_sub_list]
+
+    # update
+    description = generate_random_string()
+    tags = f"{generate_random_string()}={generate_random_string()}"
+    instance_update = run(
+        f"az iot ops update -n {instance_name} -g {resource_group} --description {description} "
+        f"--tags {tags}"
+    )
+    assert instance_update["properties"]["description"] == description
+    tag_key, tag_value = tags.split("=")
+    assert instance_update["tags"][tag_key] == tag_value
+
+
+def assert_broker_args(
+    instance_name: str,
+    resource_group: str,
+    add_insecure_listener: Optional[bool] = None,
+    broker_backend_part: Optional[str] = None,
+    broker_backend_rf: Optional[str] = None,
+    broker_backend_workers: Optional[str] = None,
+    broker_config_file: Optional[str] = None,
+    broker_frontend_replicas: Optional[str] = None,
+    broker_frontend_workers: Optional[str] = None,
+    broker_listener_type: Optional[str] = None,
+    broker_mem_profile: Optional[str] = None,
+    bfr: Optional[str] = None,
+    bfw: Optional[str] = None,
+    bp: Optional[str] = None,
+    br: Optional[str] = None,
+    bw: Optional[str] = None,
+    fr: Optional[str] = None,
+    fw: Optional[str] = None,
+    lt: Optional[str] = None,
+    mp: Optional[str] = None,
+    **_
```

Comment:
Are these args used anywhere except to optionally assign their fuller-worded counterparts?

Doesn't setting the initial vars (broker_frontend_replicas, etc) as Optional = None do this already?

Reply:
This is mostly there to make sure the shorthand arguments work still:
![image](https://github.com/user-attachments/assets/f860fbf7-4c4f-4fe9-8f6a-57973c48d737)

We could say "full args only" but I would rather just make sure every arg works

Reply:
Ok got it, so these args are direct projections of the command args passed to the CREATE_ARGS env variable